lib/runner/pre.js

```diff
@@ -0,0 +1,189 @@
+            // form data parsing simply "enriches" all form parameters having file data type by replacing / setting the
+            // value as a read stream
+            formdata: function (formdata, next) {
+                // ensure that we only process the file type
+                async.eachSeries(_.filter(formdata.all(), {type: 'file'}), function (formparam, next) {
+                    var stream;
+
+                    try {
+                        sourceReadable && formparam.src && (stream = resolver.createReadStream(formparam.src));
+                    }
+                        // eslint-disable-next-line no-empty
+                    catch (e) { } // we do not need to forward error here since next line we throw warning
+
+                    if (!stream) { // lack of stream also determines lack of `formparam.src` and source being readable
+                        triggers.console(cursor, 'warn', 'unable to load form file for upload: "' + formparam.src + '"');
+                        // @todo - do enabled: false instead of removing it
+                        formdata.remove(formparam); // remove from list since it will interfere with requester
+                    }
+                    // otherwise, replace the value with stream for making it usable in requester
+                    else {
+                        formparam.value = stream;
+                    }
+                    next();
+                }, next);
+            },
+            // file data
+            file: function (filedata, next) {
+                var stream;
+
+                try {
+                    sourceReadable && filedata.src && (stream = resolver.createReadStream(filedata.src));
+                }
+                    // eslint-disable-next-line no-empty
+                catch (e) { }
+
+                if (!stream) {
+                    triggers.console(cursor, 'warn', 'unable to load raw file for upload: "' + filedata.src + '"');
+                    filedata.value = null; // ensure this does not mess with requester
+                    delete filedata.content; // @todo - why content?
+                }
+                // stream is valid, so replace the value in data
+                else {
+                    // @todo: why is this not `value` like formdata. it would have made it easy to pass through
+                    // a common logic
+                    filedata.content = stream;
+                }
+
+                next();
+            }
+        }[mode] || async.constant()], function (err) {
+            // just as a precaution, show the error in console. each resolver anyway should handle their own console
+            // warnings.
+            // @todo - get cursor here.
+            err && triggers.console(cursor, 'warn', 'file data resolution error: ' + (err.message || err));
+            done(null); // absorb the error since a console has been trigerred
+        });
+    },
+    // Proxy lookup
+    function (payload, done) {
```

Comment:
We still need the old proxyResolver being sent like fileResolver. But this time, it will be very very simple and return OS values only.